components/feature/tabs/src/main/java/mozilla/components/feature/tabs/toolbar/TabCounterToolbarButton.kt

```diff
@@ -4,67 +4,106 @@
 
 package mozilla.components.feature.tabs.toolbar
 
-import android.view.HapticFeedbackConstants
 import android.view.View
 import android.view.ViewGroup
-import mozilla.components.browser.session.Session
-import mozilla.components.browser.session.SessionManager
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.selector.getNormalOrPrivateTabs
+import mozilla.components.browser.state.selector.normalTabs
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
 import mozilla.components.concept.toolbar.Toolbar
-import mozilla.components.feature.tabs.R
+import mozilla.components.lib.state.ext.flowScoped
 import mozilla.components.support.ktx.android.content.res.resolveAttribute
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import mozilla.components.ui.tabcounter.R
 import mozilla.components.ui.tabcounter.TabCounter
+import mozilla.components.ui.tabcounter.TabCounterMenu
 import java.lang.ref.WeakReference
 
 /**
  * A [Toolbar.Action] implementation that shows a [TabCounter].
  */
-class TabCounterToolbarButton(
-    private val sessionManager: SessionManager,
-    private val showTabs: () -> Unit
+@ExperimentalCoroutinesApi
+@Suppress("LongParameterList")
+open class TabCounterToolbarButton(
+    private val lifecycleOwner: LifecycleOwner,
+    private val filterPrivate: Boolean? = null,
+    private val showTabs: () -> Unit,
+    private val store: BrowserStore,
+    val menu: TabCounterMenu
 ) : Toolbar.Action {
-    private var reference: WeakReference<TabCounter> = WeakReference<TabCounter>(null)
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    var reference: WeakReference<TabCounter> = WeakReference<TabCounter>(null)
```

Comment:
This is only public for testing, otherwise it's private. 

I'm confused about specifying the type -- your example is the same as what I have written 🤔

Reply:
> This is only public for testing, otherwise it's private.

It can't be though because you use it in your subtype in `Fenix`. If you make it private (desired) `FenixTabCounterToolbarButton` won't compile. Ideally, we shouldn't need the subtype in Fenix, see my comments there :).

> I'm confused about specifying the type 

Ah just a copy/paste error, I updated the snipped. We don't need to specify the type here.